npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser/lockfile_parser.rb

```diff
@@ -155,31 +155,37 @@ def shrinkwrap_dependencies
         def recursively_fetch_npm_lock_dependencies(object_with_dependencies)
           dependency_set = Dependabot::NpmAndYarn::FileParser::DependencySet.new
 
-          object_with_dependencies.
-            fetch("dependencies", {}).each do |name, details|
-              next unless semver_version_for(details["version"])
+          dependencies = object_with_dependencies["dependencies"]
+          dependencies ||= object_with_dependencies.fetch("packages", {}).transform_keys do |name|
+            name.delete_prefix("node_modules/")
```

Comment:
Why does this get deleted? Also why is is fetching `"packages"` when before it didn't?

Reply:
This is the v3 format, which no longer provides `dependencies` key, and prepends `node_modules/` to packages names.